proposals/NNNN-Processing-of-unknown-enum-values-by-SDL-and-Mobile-libraries.md

```diff
@@ -0,0 +1,176 @@
+  # Processing of unknown enum values by SDL and Mobile libraries
+  * Proposal: [F129](F128-Processing-of-unknown-enum-values-by-SDL-and-Mobile-libraries.md)
+  * Author: [Igor Gapchuk](https://github.com/IGapchuk), [Dmitriy Boltovskiy](https://github.com/dboltovskyi/), [Kostiantyn Boskin](https://github.com/kostyaboss), [Yurii Lokhmatov](https://github.com/yoooriii)
+  * Status: Review
+  * Impacted platforms: [Core/iOS/JavaSuite]
+
+  ## Introduction
+The main purpose of this proposal is to gracefully handle incompatible enum values in a mobile RPC request with any given SDL Core version. This can be achieved by improving the way SDL Core processes and validates unknown enum values and by extending mobile library functionality to properly format RPC requests when RPC API’s (RAPI) are mismatched.
+
+  ## Motivation
+  
+The current filtering and data validation mechanism used by SDL Core needs be improved upon to avoid application registration issues and other RPC failures that result from differences in RAPI versions between the SDL mobile library and SDL Core.  To maintain backwards compatibility, the mobile libraries also need to be updated to format RPC requests by cutting off or ignoring unknown and unsupported enum values when a difference in RAPI sets have been detected. 
+To better understand the issue at hand, please take note of following example:
+Let us assume that a mobile application with a newer version of the RAPI attempts to connect to an SDL Core instance which is using an older RAPI version. In this example, the SDL Core RAPI version is set to 4.3 whereas the mobile application RAPI version is set to 6.0. Let us also assume that the mobile application has an AppHMIType of DEFAULT and REMOTE_CONTROL set.  The DEFAULT AppHMIType has been present in the RAPI since its inception, however REMOTE_CONTROL was introduced in version 6.0 of the RAPI.  
+
+In the case above, the mobile application will try to register itself by sending a RegisterAppInterface (RAI) request to SDL Core with an array of AppHMITypes : appHMIType[DEFAULT, REMOTE_CONTROL]. SDL Core will next try to validate the parameters in the RAI request, it will immediately find an unknown enum value of REMOTE_CONTROL. A failure response will then be sent to the mobile application with the INVALID_DATA result code (this is because the REMOTE_CONTROL AppHMIType has only been available since version RAPI 4.5, however we are communicating with a version of SDL Core that only supports RAPI 4.3). As a result of the scenario above, the mobile application fails to register with SDL Core providing an unacceptable user experience -- in addition the mobile application cannot conclusively determine the cause of failure since INVALID_DATA can be returned for any number of reasons.  Ideally the mobile app would be registered successfully with a warning provided in the result code stating some AppHMITypes could not be processed.
```

Comment:
It looks like the extra line break was added, but my other suggestions have not been implemented:
> As a result of the scenario above, the mobile application fails to register with SDL Core providing an unacceptable user experience -- in addition the mobile application cannot conclusively determine the cause of failure since INVALID_DATA can be returned for any number of reasons.  

Please change "...experience -- in addition the..." to "...experience. In addition, the..."

>Ideally the mobile app would be registered successfully with a warning provided in the result code stating "some AppHMITypes could not be processed".

Please move the period to be inside the quotation mark.